// Portions of the code are licensed under the BSD:
// MeCab -- Yet Another Part-of-Speech and Morphological Analyzer
//
//
//  Copyright(C) 2001-2006 Taku Kudo <taku@chasen.org>
//  Copyright(C) 2004-2006 Nippon Telegraph and Telephone Corporation
// All rights reserved.
//
// Redistribution and use in source and binary forms, with or without modification, are
// permitted provided that the following conditions are met:
//
//  * Redistributions of source code must retain the above
//     copyright notice, this list of conditions and the
//        following disclaimer.
//
//  * Redistributions in binary form must reproduce the above
//     copyright notice, this list of conditions and the
//        following disclaimer in the documentation and/or other
//           materials provided with the distribution.
//
//  * Neither the name of the Nippon Telegraph and Telegraph Corporation
//     nor the names of its contributors may be used to endorse or
//        promote products derived from this software without specific
//           prior written permission.
//
// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED
// WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
// PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
// ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
// LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR
// TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.

// Relicensed under the Artistic License 2.0:
// p6-MeCab -- A Perl 6 bindings for MeCab
//  Copyright(C) 2016 titsuki <titsuki@cpan.org>
// For more details, see the "LICENSE" file.
--- mecab-0.996/src/tagger.cpp	2013-02-18 01:55:30.000000000 +0900
+++ mecab-0.996-new/src/tagger.cpp	2016-11-05 16:25:51.680987399 +0900
@@ -732,7 +732,7 @@
 
 LatticeImpl::LatticeImpl(const Writer *writer)
     : sentence_(0), size_(0), theta_(kDefaultTheta), Z_(0.0),
-      request_type_(MECAB_ONE_BEST),
+      request_type_(MECAB_ONE_BEST | MECAB_ALLOCATE_SENTENCE),
       writer_(writer),
       ostrs_(0),
       allocator_(new Allocator<Node, Path>) {
